### PR TITLE
openstack: Test manifest creation

### PR DESCRIPTION
**Pull request by pierreprinetti**
Add a script for testing OpenStack manifest generation.

Co-Authored-By: Matthew Booth <mbooth@redhat.com>

### CODE REVIEW

**Comment by iamemilio** [comment]
Glad you guys picked this up! This can help us catch some usage errors that might slip through the cracks of our unit tests, and I'm excited to start playing with this.

/assign

**Comment by pierreprinetti** [comment]
> Glad you guys picked this up! This can help us catch some usage errors that might slip through the cracks of our unit tests, and I'm excited to start playing with this.
> 
> /assign

👍

Please play with it (the script should print useful error messages when used incorrectly)! Let me know if there is something to fix; this PR is a prerequisite for running [the corresponding job in the CI](https://github.com/openshift/release/pull/13950).

**Comment by mandre** [comment]
I like the idea of validating the generated manifests, this allows to increase the test coverage and prevent a whole class of regressions at low expenses in terms of CI resources.

File: hack/test-manifests-openstack.sh [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
```
Comment (staebler) [inline hack/test-manifests-openstack.sh:1]: Do you want to put this file at `hack/openstack/test-manifests.sh` so that the OpenStack team can own it?

**Review by mandre — COMMENTED** [review]
/lgtm

File: hack/test-manifests-openstack.sh (hunk 1 above)
Comment (pierreprinetti) [reply hack/test-manifests-openstack.sh]: Thank you, great suggestion. Done!

**Comment by pierreprinetti** [comment]
@staebler PTAL :)

**Review by staebler — COMMENTED** [review]
/approve

**Comment by pierreprinetti** [comment]
I’m pretty sure I can’t do this, but it would be cool to
/skip

**Pull request merged**